lite2/client.go

```diff
@@ -234,6 +234,10 @@ func (c *Client) restoreTrustedHeaderAndNextVals() error {
 		c.trustedNextVals = trustedNextVals
 
 		c.logger.Debug("Restored trusted header and next vals", lastHeight)
+	} else {
+		c.trustedHeader = nil
```

Comment:
I don't think this is the right place for nullifying the header & vals. The name of this func implies restoration, not setting trustedHeader to nil.